### PR TITLE
MAT-8752 access locked measure refactored

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-8752](https://jira.cms.gov/browse/MAT-8752)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/components/editMeasure/details/measureInformation/MeasureInformation.tsx [hunk 1]
```diff
@@ -265,11 +266,12 @@ export default function MeasureInformation(props: MeasureInformationProps) {
...
-    measure?.measureSet?.owner,
-    measure?.measureSet?.acls,
-    measure?.measureMetaData?.draft
-  );
+  const canEdit =
+    checkUserCanEdit(
+      measure?.measureSet?.owner,
+      measure?.measureSet?.acls,
+      measure?.measureMetaData?.draft
+    ) && !measureLockedByAnotherUser;
```
Comment (adongare) [inline src/components/editMeasure/details/measureInformation/MeasureInformation.tsx:274]: I think this needs to be added for cms id generation as well

Comment (sb-cecilialiu) [reply src/components/editMeasure/details/measureInformation/MeasureInformation.tsx:274]: @adongare yes you are right, I forgot it this time, I had it covered last PR. Will do thanks!

File: src/components/editMeasure/details/stewardAndDevelopers/StewardAndDevelopers.tsx [hunk 2]
```diff
@@ -53,11 +54,12 @@ export default function StewardAndDevelopers(props: StewardAndDevelopersProps) {
...
-    measure?.measureSet?.owner,
-    measure?.measureSet?.acls,
-    measure?.measureMetaData?.draft
-  );
+  const canEdit =
+    checkUserCanEdit(
+      measure?.measureSet?.owner,
+      measure?.measureSet?.acls,
+      measure?.measureMetaData?.draft
+    ) && !measureLockedByAnotherUser;
```
Comment (adongare) [inline src/components/editMeasure/details/stewardAndDevelopers/StewardAndDevelopers.tsx:62]: I'm seeing a pattern here for all the components of measure details- a lot of these `canEdits` are recalculated in individual components. we could have just passed down since MeasureDetails.tsx is calculating it anyways. not specific to your task though. it was added over the period. We can create new task if it is not quick fix

**Review by adongare — APPROVED** [review]

File: src/components/editMeasure/details/stewardAndDevelopers/StewardAndDevelopers.tsx (hunk 2 above)
Comment (sb-cecilialiu) [reply src/components/editMeasure/details/stewardAndDevelopers/StewardAndDevelopers.tsx:62]: @adongare Ok thanks! Will do more refactoring later.
